### PR TITLE
Add advanced Gitlab template

### PR DESCRIPTION
**Pull request by tabacmend**

### CODE REVIEW

**Review by lukebrogan-mend — APPROVED** [review]

**Review by samq-ws — CHANGES_REQUESTED** [review]
the scan stage must be a part of the build stage since GitLab does not share artifacts.  This example is incorrect.  @tabacmend @lukebrogan-mend  , let's discuss offline. See lines 48-50 and the link in https://github.com/mend-toolkit/mend-examples/blob/main/CI-CD/GitLab/Mend%20CLI/gitlab-mend-cli-scan-template.yml

File: CI-CD/GitLab/Mend CLI/Gitlab_template/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  extends: mend_sca
+
+.mend_reports:
+  image: python:3.9-slim-bullseye
+  stage: mend_reports
+extends: mend_reports
+```
+
+### Bonus
+By default, the template executes SCA, SAST and generate all reports which described above.
```
Comment (samq-ws) [inline CI-CD/GitLab/Mend CLI/Gitlab_template/README.md:55]: generates all reports which are described above.

File: CI-CD/GitLab/Mend CLI/Gitlab_template/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+If you wish to change it from the template or when you run the pipeline manually.
```
Comment (samq-ws) [inline CI-CD/GitLab/Mend CLI/Gitlab_template/README.md:56]: Not sure what this means.

File: CI-CD/GitLab/Mend CLI/Gitlab_template/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+In addition, all scan logs and reports are uploaded to each build stage.
```
Comment (samq-ws) [inline CI-CD/GitLab/Mend CLI/Gitlab_template/README.md:57]: Where are they uploaded?

File: CI-CD/GitLab/Mend CLI/Gitlab_template/.gitlab-ci.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
+default:
```
Comment (samq-ws) [inline CI-CD/GitLab/Mend CLI/Gitlab_template/.gitlab-ci.yaml:1]: this file is not necessary as the example should be provided in the README.md

File: CI-CD/GitLab/Mend CLI/Gitlab_template/.gitlab-ci.yaml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+#   image: maven:3.8-openjdk-11
+#   stage: build
+#   script: mvn $MAVEN_OPTS clean install $MAVEN_CLI_OPTS
+#   cache:
+#     key: ${CI_COMMIT_REF_SLUG}
+#     paths:
+#       - .m2/repository
+#       - target
+#     policy: push
+#   tags:
```
Comment (samq-ws) [inline CI-CD/GitLab/Mend CLI/Gitlab_template/.gitlab-ci.yaml:40]: These tags are not available in Gitlab.com

File: CI-CD/GitLab/Mend CLI/Gitlab_template/.gitlab-ci.yaml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+#     - docker
+  
+## NPM example - replace with your builds steps
```
Comment (samq-ws) [inline CI-CD/GitLab/Mend CLI/Gitlab_template/.gitlab-ci.yaml:43]: move to bottom of readme as shown in https://gitlab.com/samqws-oritest/templates/-/blob/main/README.md

File: CI-CD/GitLab/Mend CLI/Gitlab_template/.gitlab-ci.yaml [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+## Cache exmple for Maven
+cache:
+  key: ${CI_COMMIT_REF_SLUG}
+  paths:
+    - .m2/repository
+    - target
+  policy: pull-push
+
+## Maven example - replace with your builds steps
+# maven_build:
```
Comment (samq-ws) [inline CI-CD/GitLab/Mend CLI/Gitlab_template/.gitlab-ci.yaml:30]: uncomment and show in readme as shown in https://gitlab.com/samqws-oritest/templates/-/blob/main/README.md

**Review by samq-ws — CHANGES_REQUESTED** [review]

**Comment by tabacmend** [comment]
@samq-ws updated readme and removed the gitlab ci yaml.
pls review

**Review by samq-ws — APPROVED** [review]

**Pull request merged**